I have some concern with this PR. In a new, clean instance of AWS Smoketest it doesn't build. The reported errors are:
```
/home/centos/smoketest/PR-17195/HPCC-Platform/plugins/fileservices/fileservices.cpp:718:10: error: â€˜struct IClientSprayFixedâ€™ has no member named â€˜setDestNumPartsâ€™
/home/centos/smoketest/PR-17195/HPCC-Platform/plugins/fileservices/fileservices.cpp:852:10: error: â€˜struct IClientSprayVariableâ€™ has no member named â€˜setDestNumPartsâ€™
/home/centos/smoketest/PR-17195/HPCC-Platform/plugins/fileservices/fileservices.cpp:1028:10: error: â€˜struct IClientSprayVariableâ€™ has no member named â€˜setDestNumPartsâ€™
/home/centos/smoketest/PR-17195/HPCC-Platform/plugins/fileservices/fileservices.cpp:1164:10: error: â€˜struct IClientSprayVariableâ€™ has no member named â€˜setDestNumPartsâ€™
make[2]: *** [plugins/fileservices/CMakeFiles/fileservices.dir/build.make:118: plugins/fileservices/CMakeFiles/fileservices.dir/fileservices.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:14158: plugins/fileservices/CMakeFiles/fileservices.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

--- a/plugins/fileservices/fileservices.cpp
+++ b/plugins/fileservices/fileservices.cpp
@@ -715,6 +715,7 @@ FILESERVICES_API char * FILESERVICES_CALL implementSprayFixed(ICodeContext *ctx,
     req->setSourceRecordSize(recordSize);
     req->setDestGroup(destinationGroup);
     req->setDestLogicalName(logicalName.str());
+    req->setDestNumParts(destinationNumParts);
     req->setOverwrite(overwrite);
     req->setReplicate(replicate);
     req->setCompress(compress);
@@ -848,6 +849,7 @@ static char * implementSprayVariable(ICodeContext *ctx, const char * sourceIP, c
         req->setSourceCsvEscape(sourceCsvEscape);
     req->setDestGroup(destinationGroup);
     req->setDestLogicalName(logicalName.str());
+    req->setDestNumParts(destinationNumParts);
     req->setOverwrite(overwrite);
     req->setReplicate(replicate);
     req->setCompress(compress);
@@ -1023,6 +1025,7 @@ FILESERVICES_API char * FILESERVICES_CALL implementSprayXml(ICodeContext *ctx, c
     req->setSourceRowTag(sourceRowTag);
     req->setDestGroup(destinationGroup);
     req->setDestLogicalName(logicalName.str());
+    req->setDestNumParts(destinationNumParts);
     req->setOverwrite(overwrite);
     req->setReplicate(replicate);
     req->setCompress(compress);
@@ -1158,6 +1161,7 @@ FILESERVICES_API char * FILESERVICES_CALL implementSprayJson(ICodeContext *ctx, 
     req->setSourceRowTag(sourceRowPath);
     req->setDestGroup(destinationGroup);
     req->setDestLogicalName(logicalName.str());
+    req->setDestNumParts(destinationNumParts);
     req->setOverwrite(overwrite);
     req->setReplicate(replicate);
     req->setCompress(compress);

